src/firstProblem.cpp

```diff
@@ -0,0 +1,236 @@
+	acceptance_coefficient = _acceptance_coefficient;
+
+	// random seed is set to a default value, for reproducibility
+	rng.seed(1);
+}
+
+void SimulatedAnnealing::get_initial_solution(double x[]) {
+	// employ theoretic bound to get a suitable starting point
+	// x value has already been proven valid in superclass
+	TheoreticBound::run_search(x);
+}
+
+void SimulatedAnnealing::get_neighbour(double x[], double new_x[]) {
+	// copy x to new array before perturbation
+	for(int d=0; d<K; d++) {
+		new_x[d] = x[d];
+	}
+
+	/**
+	* ------------------------
+	* ### Algorithm
+	* Find new point candidate modifying previous solution.
+	* component \f$ d \sim \mathcal{U}[1, K] \f$ of \f$ \vec{x} \f$ is perturbed
+	* summing it a quantity \f$ \xi \sim \mathcal{U}[-1, 1] \f$
+	*/
+	std::uniform_real_distribution<double> perturbation(-50, 50);
+	std::uniform_int_distribution<int> index_choice(0, K-1);
+	int chosen_d = index_choice(rng);
+	new_x[chosen_d] = x[chosen_d] + perturbation(rng);
+
+	/** Stop if perturbation jumps inside valid solutions space,
+		otherwise perturb \f$ \vec{x} \f$ again */
+	if( !respect_constraints(new_x) ) {
+		get_neighbour(x, new_x);
+	}
+}
+
+double SimulatedAnnealing::acceptance_probability(double x[], double new_x[]) {
+	double delta = objective_function(new_x) - objective_function(x);
+	/**
+	* ---------------
+	* ### Algorithm
+	* - accept better solutions with probability 1
+	*/
+	if(delta <= 0){
+		return 1.0;
+	}
+	/**
+	* - accept worse solutions with probability \f$ e^{- \frac{acceptance_coefficient \Delta}{T} } \f$
+	*/
+	return exp(-acceptance_coefficient * delta / temperature);
+}
+
+double SimulatedAnnealing::new_temperature() {
+	// exponential decay of temperature
+	return temperature * cooling_rate;
+}
+
+double SimulatedAnnealing::temperature_steps() {
+	return steps_coefficient / temperature;
+}
+
+void SimulatedAnnealing::run_search(double x[]) {
+	get_initial_solution(x);
+
+	// keep trace of how much jumps have been done since the beginning
+	int current_iteration = 0;
+
+	// save current best result in this variable
+	double* best_x = (double*) malloc(K * sizeof(double));
+	double best_score = std::numeric_limits<double>::infinity();
+
+	// store here the best value of x when swapping old with new one (see below)
+	double* tmp; // = (double*) malloc(K * sizeof(double));
+
+	// store here new point explored with its score
+	double* new_x = (double*) malloc(K * sizeof(double));
+	double new_score;
+
+	// random variable that set the threshold for accepting worse solutions
+	std::uniform_real_distribution<double> acceptance_threshold(0,1);
+
+	while(current_iteration <= max_iterations && temperature > 0.5) {
+		// round of search for current temperature
+		std::cout << "Temperature " << temperature << " at iteration " <<
+			current_iteration << "/" << max_iterations << "\n";
+		std::cout << "Best score: " << best_score << "\n";
+
+		// keep trace of mean value of acceptance_probability when new point is worse
```

Comment:
In slides they said that acceptance_probability of the wrong matches goes to zero with the temperature, I've kept trace of it for debug